src/components/TodoElement.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  removeTodo: (id: number) => void;
+  loadingTodo: number[];
+  updateStatus: (id: number) => void;
+  updateTitle: (id: number, title: string) => void;
+}
+
+export const TodoElement: React.FC<Props> = ({
+  todo,
+  removeTodo,
+  loadingTodo,
+  updateStatus,
+  updateTitle,
+}) => {
+  const { id, title, completed } = todo;
+  const [editValue, setEditValue] = useState(title);
+  const [editForm, setEditForm] = useState(false);
+  const [initialEditValue, setInitialEditValue] = useState('');
+
+  const ref = useRef<HTMLInputElement>(null);
+
+  const handleEditValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditValue(event.target.value);
+  };
+
+  useEffect(() => {
+    ref.current?.focus();
+  }, [editForm]);
+
+  const applyChanges = () => {
+    updateTitle(id, editValue);
+    setEditForm(false);
+  };
+
+  const handleDoubleClick = () => {
+    setInitialEditValue(editValue);
+    setEditForm(true);
+  };
+
+  const handleBlur = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!editValue.trim()) {
+      removeTodo(id);
+    }
+
+    applyChanges();
+  };
+
+  const cancelEditing = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setEditValue(initialEditValue);
+      setEditForm(false);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!editValue) {
+      removeTodo(id);
+    }
+
+    if (editValue === title) {
+      setEditValue(title);
+      setEditForm(false);
```

Comment:
you do setEditForm(false); inside of applyChanges function, which executes anyway, are you sure you need it here?